src/components/async-select.js

```diff
@@ -0,0 +1,31 @@
+// outsource dependencies
+import React from 'react';
+import PropTypes from 'prop-types';
+import AsyncSelect from 'react-select/async';
+
+function CustomAsyncSelect ({ name='roleSelect', label, valueKey='name', labelKey='name', onChange, ...rest }) {
+  return <AsyncSelect
+    id={name}
+    name={name}
+    defaultOptions
+    cacheOptions={false}
+    isDisabled={rest.disabled}
+    onChange={event => onChange(event)}
```

Comment:
```suggestion
    onChange={onChange}
```

Can I write it so? what is the difference?

Reply:
> Can I write it so? what is the difference?

You can. The difference is that in the first case, I pass the arrow function to the component that will call onChange, and in the second case the function onChange just passed to  the component. I guess in this case an additional layer is not required.